Sources/PackageDescription/PackageDescription.swift

```diff
@@ -290,6 +290,50 @@ public final class Package {
         registerExitHandler()
     }
     
+    /// Initializes a Swift package with configuration options you provide.
+    ///
+    /// - Parameters:
+    ///   - name: The name of the Swift package, or `nil` to use the package's Git URL to deduce the name.
+    ///   - defaultLocalization: The default localization for resources.
+    ///   - platforms: The list of supported platforms with a custom deployment target.
+    ///   - pkgConfig: The name to use for C modules. If present, Swift Package Manager searches for a
+    ///   `<name>.pc` file to get the additional flags required for a system target.
+    ///   - providers: The package providers for a system target.
+    ///   - products: The list of products that this package makes available for clients to use.
+    ///   - dependencies: The list of package dependencies.
+    ///   - targets: The list of targets that are part of this package.
+    ///   - cLanguageStandard: The C language standard to use for all C targets in this package.
+    ///   - cxxLanguageStandard: The C++ language standard to use for all C++ targets in this package.
+    @available(_PackageDescription, introduced: 5.3)
+    @available(_PackageDescription, deprecated: 6, renamed:"init(name:defaultLocalization:platforms:pkgConfig:providers:products:dependencies:targets:swiftLanguageModes:cLanguageStandard:cxxLanguageStandard:)")
+    public init(
```

Comment:
I think we can get the same effect by removing this overload and using `@_disfavoredOverload` on the old one, fewer overloads is usually easier to deal with, do you have a strong reference with this one?

Reply:
I didn't realize that `@_disfavoredOverload` was acceptable in public API.

But since it seems to be acceptable, I think `@_disfavoredOverload` would be a better way to go.

It would let us avoid adding a completely new init method just to have the overloads work out right.

Reply:
Yeah, it's fine to use responsibly :)